auth/proxy/app.ts

```diff
@@ -0,0 +1,150 @@
+import { APIGatewayProxyResultV2, APIGatewayProxyEventV2, APIGatewayProxyEventHeaders } from 'aws-lambda';
+import https from 'https';
+import querystring from 'querystring';
+
+const ALLOWED_ENDPOINTS = [
+  "authorize",
+  "token",
+  '/.well-known/openid-configuration'
+]
+
+const rejectUnauthorisedEndpoints = (path: string) => {
+  const isAllowedEndpoint = !ALLOWED_ENDPOINTS.includes(path)
+
+  if (!isAllowedEndpoint) {
+    throw new Error('Endpoint is not whitelisted.')
+  }
+}
+
+// cognito expects consistent casing for header names e.g. x-amz-target
+// host must be removed to avoid ssl hostname unrecognised errors
+const sanitizeHeaders = (headers: APIGatewayProxyEventHeaders) => {
+  return Object.entries(headers)
+    .filter(([key]) => key.toLowerCase() !== 'host')
+    .reduce<Record<string, string>>((acc, [key, value]) => {
+      acc[key.toLowerCase()] = value || '';
+      return acc;
+    }, {});
+}
+
+const transformCognitoUrl = (url: string | undefined) => {
+  return url?.toLowerCase().replace('_', '')
```

Comment:
Cognito `UserPoolId` Cloudformation reference adds an extra underscore - can we use a different reference and remove this completely?